### PR TITLE
Update external build-deps tools paths (MBE)

### PR DESCRIPTION
**Pull request by ashwinimurt**
To support external repo mono-build-deps PR: https://ono.unity3d.com/unity-extra/mono-build-deps/pull-request/77510/_/scripting/move-zip-files-into-folders-hgmove

Katana:
https://katana.bf.unity3d.com/projects/Mono%20Runtime%20and%20Classlibs/builders?boo_branch=unity-trunk&cecil_branch=unity-master&krait-signal-handler_branch=master&mono_branch=support-rearranged-mono-build-deps&mono-build-tools-extra_branch=master&mono-build-deps_branch=scripting%2Fmove-zip-files-into-folders&unityscript_branch=unity-trunk

### CODE REVIEW

File: external/buildscripts/build.pl [hunk 1]
```diff
@@ -351,17 +351,17 @@
 		my $texinfoVersion = "4.8";
 		my $automakeVersion = "1.15";
 		my $libtoolVersion = "2.4.6";
-		my $autoconfDir = "$externalBuildDeps/autoconf-$autoconfVersion";
-		my $texinfoDir = "$externalBuildDeps/texinfo-$texinfoVersion";
-		my $automakeDir = "$externalBuildDeps/automake-$automakeVersion";
-		my $libtoolDir = "$externalBuildDeps/libtool-$libtoolVersion";
+		my $autoconfDir = "$externalBuildDeps/autoconf-2-69/autoconf-$autoconfVersion";
```
Comment (ashwinimurt) [inline external/buildscripts/build.pl]: Stevedore does not allow `.` in the unpacked folder name. So use dashes instead.

File: external/buildscripts/build_us_and_boo.pl [hunk 2]
```diff
@@ -49,24 +49,17 @@ sub BuildUnityScriptFor45
...
 	
-	# Build host is handling this
-	if (!$buildMachine)
+	if (!(-d "$booCheckout"))
 	{
-		if (!(-d "$booCheckout"))
-		{
-			print(">>> Checking out boo\n");
-			GitClone("git://github.com/Unity-Technologies/boo.git", $booCheckout, "unity-trunk");
-		}
```
Comment (ashwinimurt) [inline external/buildscripts/build_us_and_boo.pl]: Needed for Gitlab CI builds

File: external/buildscripts/build_unityscript_bareminimum_win.pl [hunk 3]
```diff
@@ -65,19 +65,14 @@ sub UnityBooc
 
 sub BuildUnityScriptForUnity
 {	
-	# Build system is handling this
-	if (!$ENV{UNITY_THISISABUILDMACHINE}) {
-		GitClone("git://github.com/Unity-Technologies/boo.git", $booCheckout);
-	}
+	GitClone("git://github.com/Unity-Technologies/boo.git", $booCheckout);
```
Comment (joncham) [inline external/buildscripts/build_unityscript_bareminimum_win.pl]: Do we always want to do this now?

File: external/buildscripts/build_unityscript_bareminimum_win.pl [hunk 4]
```diff
@@ -65,19 +65,14 @@ sub UnityBooc
...
 
 	my $commonDefines = "NO_SERIALIZATION_INFO,NO_SYSTEM_PROCESS,NO_ICLONEABLE,MSBUILD,IGNOREKEYFILE";
 
 	Build("$booCheckout/src/booc/Booc.csproj", undef, "/property:TargetFrameworkVersion=4.0 /property:DefineConstants=\"" . $commonDefines . "\" /property:OutputPath=$output/wp8");
 	Build("$booCheckout/src/booc/Booc.csproj", undef, "/property:TargetFrameworkVersion=4.0 /property:DefineConstants=\"" . $commonDefines . ",NO_SYSTEM_REFLECTION_EMIT\" /property:OutputPath=$output/wsa");
 	
-	if (!$ENV{UNITY_THISISABUILDMACHINE}) {
-		GitClone("git://github.com/Unity-Technologies/unityscript.git", $usCheckout);
-	}
+	GitClone("git://github.com/Unity-Technologies/unityscript.git", $usCheckout);
```
Comment (joncham) [inline external/buildscripts/build_unityscript_bareminimum_win.pl]: Same here

File: external/buildscripts/build_unityscript_bareminimum_win.pl (hunk 3 above)
Comment (ashwinimurt) [reply external/buildscripts/build_unityscript_bareminimum_win.pl]: Its needed for OSX and bare minimum builds on Gitlab CI

**Pull request merged**